src/CoffeeShop/CoffeeShopWithDistanceToUser.php

```diff
@@ -0,0 +1,48 @@
+<?php
+namespace CoffeeShop;
+use base\ILocation;
+
+class CoffeeShopWithDistanceToUser extends CoffeeShop{
+    private $__distance;
+    private $__userLocation;
+
+    public function __construct($xCoord, $yCoord, $shopName)
+    {
+        parent::__construct($xCoord, $yCoord, $shopName);
+        $this->__userLocation = NULL;
+    }
+
+    private function setDistance($distance){
+        if(is_float($distance)){
+            $this->__distance = $distance;
+        }
+        return $this;
+    }
+
+    public function getDistance() : float{
+        $distance = 0.0;
+        if($this->__userLocation instanceof ILocation){
+            $distance = $this->calcDistance();
+        }
+        else{
+            throw new CoffeeShopException("User location not set. Cannot calculate and get distance.");
```

Comment:
Won't it be a better idea to throw this exception when the user location is set?